Tests/Indicators/HilbertTransformIndicatorTests.cs

```diff
@@ -0,0 +1,71 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using NUnit.Framework;
+using QuantConnect.Indicators;
+
+namespace QuantConnect.Tests.Indicators;
+
+[TestFixture]
+public class HilbertTransformIndicatorTests : CommonIndicatorTests<IndicatorDataPoint>
+{
+    protected override IndicatorBase<IndicatorDataPoint> CreateIndicator()
+    {
+        return new HilbertTransformIndicator();
+    }
+
+    protected override string TestFileName => "spy_with_hilbert.csv";
```

Comment:
What is the source of this file? Did you generate it or can we find its content at  http://www.technicalanalysis.org.uk/moving-averages/Ehle.pdf?

Reply:
I did see that J.Ehlers provides data that can be used for testing. I generated it myself from the code https://github.com/ooples/OoplesFinance.StockIndicators/blob/be83f40b547301aae7d0196b1255f4eed38d61e9/src/Calculations/Ehlers.cs#L3781-L3823

Reply:
I believe LS shared values for each indicator to assert against at https://github.com/QuantConnect/Lean/issues/6986#issuecomment-1441554170 should use these 👌